### PR TITLE
Improve login flow

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/19476

- Changes UI
- Adds provider links to the password form
- Adds a link to reset user password
- Converts to typescript and styled-components
- Adds tests

Screenshot 2021-12-29 at 17 36 42
Screenshot 2021-12-29 at 17 36 29
Screenshot 2021-12-29 at 17 36 49
Screenshot 2021-12-29 at 17 37 07

How to test:
- Signing in and out with password
- Setup Google auth by following the instructions at Admin > Authentication
- Signing in with Google auth should work

### CODE REVIEW

File: frontend/src/metabase/entities/users/forms.js [hunk 1]
```diff
@@ -132,7 +164,7 @@ export default {
       {
         name: "email",
         title: t`Email address`,
-        placeholder: t`The email you use for your Metabase account`,
+        placeholder: "The email you use for your Metabase account",
```
Comment (alxnddr) [inline frontend/src/metabase/entities/users/forms.js:167]: Why did you remove l10n here?

**Review by alxnddr — APPROVED** [review]
Works as expected, please check one comment about l10n before merging

File: frontend/src/metabase/entities/users/forms.js (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/entities/users/forms.js:167]: Not intended, fixing
